extensions/qute/qute-reference-guide.adoc

```diff
@@ -0,0 +1,673 @@
+First, all available `NamespaceResolver` s will be used to find the current context object.
+And afterwards value resolvers will be used to resolve "colors" against the context object found. 
+
+----
+{name} <1>
+{item.name} <2>
+{global:colors} <3> 
+----
+<1> no namespace, one part -`name`
+<2> no namespace, two parts - `item`, `name`
+<3> namespace `global`, one part - `colors`
+
+An expression part could be a "virtual method" in which case the name can be followed by parameters in parentheses. 
+
+----
+{item.getLabels(1)} <1>
+{name or 'John'} <2>
+----
+<1> no namespace, two parts - `item`, `getLabels(1)`, the second part is a virtual method with name `getLabels` and params `1`
+<2> infix notation, translated to `name.or('John')`; no namespace, two parts - `name`, `or('John')`
+
+[[current_context_object]]
+===== Current Context
+
+If an expression does not specify a namespace the current context object is derived from the position of the tag.
+By default, the current context object represents the data passed to the template instance.
+However, sections may change the current context object.
+A typical example is for/each loop - during iteration the content of the section is rendered with each element as the current context object:
+
+----
+{#each items}
+ {count}. {it.name} <1>
+{/each}
+
+{! Another form of iteration... !}
+{#for item in items}
+  {count}. {item.name} <2>
+{/for}
+----
+<1> `it` is an implicit alias. `name` is resolved against the current iteration element.
+<2> Loop with an explicit alias `item`.
+
+[TIP]
+====
+Data passed to the template instance are always accessible using the `data` namespace. 
+This could be useful to access data for which the key is overrided:  
+
+[source,html]
+----
+<html>
+{item.name} <1>
+<ul>
+{#for item in item.getDerivedItems()} <2>
+  <li>
+  {item.name} <3> 
+  is derived from
+  {data:item.name} <4>
+  </li>
+{/for}
+</ul>
+</html>
+----
+<1> `item` is passed to the template instance as a data object.
+<2> Iterate over the list of derived items.
+<3> `item` is an alias for the iterated element.
+<4> Use the `data` namespace to access the `item` data object.
+
+====
+
+==== Sections
+
+A section:
+
+* has a start tag
+** starts with `#`, followed by the name of the section such as `{#if}` and `{#each}`,
+* may be empty 
+** tag ends with `/`, ie. `{#emptySection /}`
+* may contain other expression, sections, etc.
+** the end tag starts with `/` and contains the name of the section (optional): `{#if foo}Foo!{/if}` or `{#if foo}Foo!{/}`,
+
+The start tag can also define parameters.
+The parameters have optional names.
+A section may contain several content *blocks*. 
+The "main" block is always present.
+Additional/nested blocks also start with `#` and can have parameters too - `{#else if item.isActive}`.
+A section helper that defines the logic of a section can "execute" any of the blocks and evaluate the parameters.
+
+----
```

Comment:
FTR, Asciidoc behaves a bit differently when you define block like that and proper `[source]`. I recommend to always add `[source]`.